sdk/compute/Azure.ResourceManager.Compute/src/autorest.md

```diff
@@ -101,16 +111,59 @@ directive:
   - from: swagger-document
     where: $.definitions.DiskSecurityType["x-ms-enum"].values[3]
     transform: $["name"] = "ConfidentialVmDiskEncryptedWithCustomerKey"
+  - from: skus.json
+    where: $.definitions
+    transform: >
+      $.ResourceSku["x-ms-client-name"] = "ComputeResourceSku";
+      $.ResourceSkuCapacity["x-ms-client-name"] = "ComputeResourceSkuCapacity";
   - from: compute.json
-    where: $.definitions.VirtualMachineInstallPatchesParameters.properties.maximumDuration
-    transform: $["format"] = "duration"
-  - from: compute.json
-    where: $.definitions.VirtualMachineExtensionUpdateProperties.properties.type
-    transform: $["x-ms-client-name"] = "VirtualMachineExtensionType"
-  - from: communityGallery.json
-    where: $.definitions.PirCommunityGalleryResource.properties.type
-    transform: $["x-ms-client-name"] = "ResourceType"
+    where: $.definitions
+    transform: >
+      $.VirtualMachineImageProperties.properties.dataDiskImages.description = "The list of data disk images information.";
+      $.VirtualMachineInstallPatchesParameters.properties.maximumDuration["format"] = "duration";
+      $.VirtualMachineExtensionUpdateProperties.properties.type["x-ms-client-name"] = "ExtensionType";
+      $.VirtualMachineExtensionProperties.properties.type["x-ms-client-name"] = "ExtensionType";
+      $.VirtualMachineScaleSetExtensionProperties.properties.type["x-ms-client-name"] = "ExtensionType";
+      $.VirtualMachineScaleSetExtension.properties.type["x-ms-format"] = "resource-type";
+      $.VirtualMachineNetworkInterfaceIPConfigurationProperties.properties.privateIPAddressVersion["x-ms-enum"].name = "IPVersion";
+      $.VirtualMachinePublicIPAddressConfigurationProperties.properties.publicIPAddressVersion["x-ms-enum"].name = "IPVersion";
+      $.SubResource["x-ms-client-name"] = "ComputeWriteableSubResourceData";
+      $.SubResource.properties.id["x-ms-format"] = "arm-id";
```

Comment:
rename the `id`?

Reply:
No, we cannot rename the id property because this is designed to use for various resource references, including public ip, network interface, subnet, restorePointGroup, etc...
The reason why this model still is here, is that some model is inheriting from it. Therefore maybe "Sub"Resource is not a great name here.
But we could see what arch board will say about this...